presto-tests/src/test/java/io/prestosql/sql/planner/planPrinter/TestPlanPrinterUtil.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner.planPrinter;
+
+import io.prestosql.Session;
+import io.prestosql.metadata.Metadata;
+import io.prestosql.metadata.OperatorNotFoundException;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.block.IntArrayBlock;
+import io.prestosql.spi.block.MethodHandleUtil;
+import io.prestosql.spi.type.ArrayType;
+import io.prestosql.spi.type.IntegerType;
+import io.prestosql.spi.type.MapType;
+import io.prestosql.spi.type.Type;
+import io.prestosql.tests.AbstractTestQueryFramework;
+import io.prestosql.tests.TestLocalQueries;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.prestosql.block.BlockAssertions.createLongsBlock;
+import static io.prestosql.block.BlockAssertions.createStringsBlock;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.VarcharType.VARCHAR;
+import static io.prestosql.util.StructuralTestUtil.mapType;
+
+public class TestPlanPrinterUtil
+        extends AbstractTestQueryFramework
```

Comment:
This doesn't need to extend AbstractTestQueryFramework. That's intended for writing tests that will execute every query against a custom query runner (e.g., to test different connectors or configurations of the query runner).

Reply:
Also, I think these tests are a little brittle (one of them relies on an implementation detail of the map type) and overkill, as they are mostly testing whether cast to varchar works. I'd just remove them altogether.

Reply:
I can remove these tests, no problem, but the reason I added these in the first place was to make sure that the castToVarchar fails for non primitive types. The function requires a Session and FunctionRegistry and I wasn't sure how to get those without using the TestLocalQueries::createLocalQueryRunner. I'm sure there's a better way, though. WDYT?

Reply:
The issue is that it should fail for any type that doesn't have a cast to varchar. Some complex types may implement a cast. The right way to test this would be to create a synthetic type just for this purpose, but again, I think that's overkill.

Reply:
Ok, I'll just remove this test then.